docs/spec_draft.md

```diff
@@ -3080,6 +3124,50 @@ More formally, for each element `x`: `real(x) = is_complex(x) ? x.real : x`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.recv
+
+### Semantics
+
+Receives data into a partition from a [stablehlo.send](#stablehlosend)
+instruction, in another partition, that shares the same channel handle pair
+{`channel_id`, `channel_type`}.
+
+If `is_host_transfer` is `true`, then the operation transfers data from the
+host.
```

Comment:
Proposed rewording:

```
Receives data from a channel with `channel_id` and produces `results`.

If `is_host_transfer` is `true`, then the operation transfers data from the
host. Otherwise, it transfers data from another device. What this means is
implementation-defined.

`results` consist of payload values which come first and a token which comes
last. The operation produces a token to reify the side effect of this operation 
as  a value that other operations can take a data dependency on.
```

Feedback:
  1. In my opinion, the current formalism goes into too much detail that isn't strictly needed to spec the op.
  2. "into a partition [...] in another partition". This can all be replaced by "a channel". In the parallel semantics PR, we intentionally didn't go into the details of what a replica and a partition are because it wasn't required, and I don't think it's required here either.
  3. "from a stablehlo.send instruction". Let's leave the formalism of who can read from channels and write to channels to future work. This formulation in the PR invites many questions that "from a channel" abstracts us away from.
  4. "the same channel handle pair {`channel_id`, `channel_type`}". As discussed above, I don't think `ChannelHandle` should be a think in StableHLO, hence the recommendation to drop this.
  5. Expanded the second paragraph to provide more details.
  6. Called out the token produced by the op and its semantics.